File: Orders.Models/Entities/Order.cs
using System.ComponentModel.DataAnnotations;

namespace Orders.Domain.Entities
{
    public class Order
    {
        [Key]
        public Guid OrderId { get; set; } = Guid.NewGuid();
        [Required]
        public string OrderNumber { get; set; } = string.Empty;

        [Required]
        public Guid CustomerId { get; set; }
        [Required]
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;

        [Required]
        public decimal TotalAmount { get; set; }

        public Customer Customer { get; set; } = null!;
        public ICollection<OrderDetail> OrderDetails { get; set; } = new List<OrderDetail>();



    }
}

File: OrdersAPI/ClientApp/OrderAngularApp/src/app/model/order.ts
import { OrderDetail } from "./order-detail"; // Import OrderDetail model
import { Customer } from "./customer"; // Import Customer model

export class Order {
  orderId: string;
  orderNumber: string;
  customerId: string;
  orderDate: Date;
  totalAmount: number;
  customer?: Customer; // Optional property for related customer data
  orderDetails: OrderDetail[]; // List of order details

  constructor(
    orderId: string,
    orderNumber: string,
    customerId: string,
    orderDate: Date,
    totalAmount: number,
    orderDetails: OrderDetail[],
    customer?: Customer
  ) {
    this.orderId = orderId;
    this.orderNumber = orderNumber;
    this.customerId = customerId;
    this.orderDate = orderDate;
    this.totalAmount = totalAmount;
    this.orderDetails = orderDetails;
    this.customer = customer;
  }
}
